config.py

```diff
@@ -0,0 +1,70 @@
+from irc3 import utils
+from decouple import config, Csv
+
+
+class Config(utils.Config):
+    """Configuration settings.
+
+    All user defined configurations are
+    loaded from environment variables.
+    """
+    def __init__(self, *args, **kwargs):
+        """Initializes configuration settings."""
+        super(Config, self).__init__(*args, **kwargs)
+        self.update(**{
+            'host': config('IRC_HOST', default='localhost'),
+            'port': config('IRC_PORT', default=6667, cast=int),
+            'nick': config('IRC_NICK', default='mybot'),
+            'username': config('IRC_USERNAME', default='mybot'),
+            'autojoins': config('IRC_AUTOJOINS', default='', cast=Csv()),
+            'database': config('DATABASE_URL', default='sqlite:///:memory:'),
+            'includes': [
```

Comment:
What irc3's parser_config does is just read the ini file and convert in this same dictionary I've called config. So whatever you would do in the ini file you can do here too.

The idea was just separate what we think is mutable, or user defined, and what is our "core", like what plugins are required for example, and what are optional. Anything else should've to be managed by changing the source code.

It's possible to do what you've asked, the possibility of load some extra plugins using env vars.

Another possibility is to rollback one or two changes I've made and keep the settings.ini file, just in order to set the plugins list and theirs respective  settings, but I need to say that the idea of having so much configuration files bothers me.

And I definitely don't want to let everything in the settings.ini file the way it was because if I want to run it in any cloud platform I would have to send that file with some sensitive information along side the source code.

Let me do the changes that loads the plugins settings from an ini file and them we can discuss again. I think it's better when we have some code to read and point. :)

Reply:
@bmeneguele take a look now, it was my first approach but I've decided to change it in order to keep just one file to manipulate, but I agree that it would make some basic configurations way more complex.

This solution basically loads a default settings that we must always keep it untouched and changing carefully because defines the mandatory  settings and plugins.

Then, it looks for a local settings file, if it exists it'll be loaded and override the default settings.

Finally it looks for some environment variables, those who were defined will also override the settings.

What do you think?

Reply:
Yes, I think it's better to keep both files. I understand and agree it's not intuitive, but some configurations just need a better way to be stored (.ini file in this case). I'm ok with this solution and the loading precedence, that's something we need to highlight in the docs when we start the work on that.